samples/excel/pivot-tables/refresh-pivot-table.yaml

```diff
@@ -0,0 +1,166 @@
+
+                // Run all of the queued up commands
+                await context.sync();
+            }
+            catch (error) {
+                 OfficeHelpers.Utilities.log(error);
+            }
+        })
+    }
+
+
+    /** Refresh pivot table in the sheet */
+    function refreshPivotTable() {
+        Excel.run(async (context) => {
+            let sheet: Excel.Worksheet;
+            try {
+                // Queue a command to get the sheet
+                sheet = context.workbook.worksheets.getActiveWorksheet();
+
+                // Call the helper to sort the table
+                await queueRefreshPivotTable(sheet);
+                
+                // Run all of the queued up commands
+                await context.sync();
+            }
+            catch (error) {
+                 OfficeHelpers.Utilities.log(error);
+            }
+        })
+    }
+
+    /** Create the table */ 
+    async function queueCreateTable(sheet: Excel.Worksheet) {
+
+            // Queue a command to add a table to the sheet
+            let expensesTable = sheet.tables.add('A1:D1', true);
+            expensesTable.name = "ExpensesTable";
+
+            // Queue a command to set the header row
+            expensesTable.getHeaderRowRange().values = [["DATE", "MERCHANT", "CATEGORY", "AMOUNT"]];
+
+            // Queue a command to add rows to the table
+            expensesTable.rows.add(null, [
+                ["1/1/2017", "THE PHONE COMPANY", "Communications", "$120"],
+                ["1/2/2017", "NORTHWIND ELECTRIC CARS", "Transportation", "$142"],
+                ["1/5/2017", "BEST FOR YOU ORGANICS COMPANY", "Groceries", "$27"],
+                ["1/10/2017", "COHO VINEYARD", "Restaurant", "$33"],
+                ["1/11/2017", "BELLOWS COLLEGE", "Education", "$350"],
+                ["1/15/2017", "TREY RESEARCH", "Other", "$135"],
+                ["1/15/2017", "BEST FOR YOU ORGANICS COMPANY", "Groceries", "$97"]
+            ]);  
+
+            //Autofit columns and rows
+            sheet.getUsedRange().getEntireColumn().format.autofitColumns();
+            sheet.getUsedRange().getEntireRow().format.autofitRows();
+        }
+
+        // Refresh pivot table
+        async function queueRefreshPivotTable(sheet: Excel.Worksheet) {
+
+            // Queue a command to get the table
+            const expensesPivot = sheet.pivotTables.getItem("PivotTable1");
+
+            // Make a change in the data
+            const expensesTable = sheet.tables.getItem("ExpensesTable");
+
+            let range = expensesTable.columns.getItem("AMOUNT").getDataBodyRange().getRow(2).values = [["$500"]];
+
+            // Run all of the queued up commands
+            await sheet.context.sync();
+
+            // Queue a command to refresh the pivot table 
+            expensesPivot.refresh();
+
+        }
+
+    /** Helper to delete the Playground sheet if already exists */
+    function deleteIfSheetExists(sheetname:string){
```

Comment:
Make into 1.1 compatible:

imagine we have:  OfficeHelpers.Excel.deleteWorksheetIfExists(name): Promise<void>

The helper function would do a requirements check and do 1.1 compatible for old, 1.4 for new